### PR TITLE
fix(pt): finetuning property/dipole/polar/dos fitting with multi-dimensional data causes error

### PR DESCRIPTION
**Pull request by Chengqian-Zhang**
Fix issue #4108 

If a pretrained model is labeled with energy and the `out_bias` is one dimension. If we want to finetune a dos/polar/dipole/property model using this pretrained model, the `out_bias` of finetuning model is multi-dimension(example: numb_dos = 250). An error occurs: 
`RuntimeError: Error(s) in loading state_dict for ModelWrapper:`
`       size mismatch for model.Default.atomic_model.out_bias: copying a param with shape torch.Size([1, 118, 1]) from checkpoint, the shape in current model is torch.Size([1, 118, 250]).`
`        size mismatch for model.Default.atomic_model.out_std: copying a param with shape torch.Size([1, 118, 1]) from checkpoint, the shape in current model is torch.Size([1, 118, 250]).`

When using new fitting, old out_bias is useless because we will recompute the new bias in later code. So we do not need to load old out_bias when using new fitting finetune.

## Summary by CodeRabbit

- **New Features**
	- Enhanced parameter collection for fine-tuning, refining criteria for parameter retention.
	- Introduced a model checkpoint file for saving and resuming training states, facilitating iterative development.

- **Tests**
	- Added a new test class to validate training and fine-tuning processes, ensuring model performance consistency across configurations.

### CODE REVIEW

File: deepmd/pt/train/training.py [hunk 1]
```diff
@@ -484,7 +484,11 @@ def collect_single_finetune_params(
                             if i != "_extra_state" and f".{_model_key}." in i
                         ]
                         for item_key in target_keys:
-                            if _new_fitting and ".fitting_net." in item_key:
+                            if _new_fitting and (
+                                (".fitting_net." in item_key)
+                                or (".out_bias" in item_key)
+                                or (".out_std" in item_key)
```
Comment (njzjz) [inline deepmd/pt/train/training.py:490]: I think it should be `".descriptor" not in item_key`. When the model has other variables in the future, is it expected to keep it or replace it?

Comment (Chengqian-Zhang) [reply deepmd/pt/train/training.py:490]: In current version, the two ways of writing `".descriptor" not in item_key` and `(".fitting_net." in item_key) or (".out_bias" in item_key) or (".out_std" in item_key)` are fully equivalent. Please @iProzd  take a look at this.

Comment (iProzd) [reply deepmd/pt/train/training.py:490]: I think `".descriptor." not in item_key` is better. Yet @Chengqian-Zhang please check and ensure that they are equivalent.

Comment (Chengqian-Zhang) [reply deepmd/pt/train/training.py:490]: image

Comment (Chengqian-Zhang) [reply deepmd/pt/train/training.py:490]: I an sure that in current version, if "descriptor" and "fitting" do not in target_keys, only "out_bias" and "out_std" remain.

**Review by njzjz — APPROVED** [review]

**Review by wanghan-iapcm — APPROVED** [review]

**Review by iProzd — APPROVED** [review]

**Pull request merged**